pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -0,0 +1,61 @@
+package core
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+func Test_genericScheduler_AssignReplicas(t *testing.T) {
+	tests := []struct {
+		name      string
+		clusters  []*clusterv1alpha1.Cluster
+		placement *policyv1alpha1.Placement
+		object    *workv1alpha2.ResourceBindingSpec
+		want      []workv1alpha2.TargetCluster
+		wantErr   bool
+	}{
+		{
+			name: "replica 4, dynamic weighted 1:1",
+			clusters: []*clusterv1alpha1.Cluster{
+				helper.NewClusterWithResource(ClusterMember1, corev1.ResourceList{
+					corev1.ResourcePods: *resource.NewQuantity(4, resource.DecimalSI),
+				}, util.EmptyResource().ResourceList(), util.EmptyResource().ResourceList()),
+				helper.NewClusterWithResource(ClusterMember2, corev1.ResourceList{
+					corev1.ResourcePods: *resource.NewQuantity(4, resource.DecimalSI),
+				}, util.EmptyResource().ResourceList(), util.EmptyResource().ResourceList()),
+			},
+			object: &workv1alpha2.ResourceBindingSpec{
+				Replicas: 4,
+			},
+			placement: &policyv1alpha1.Placement{
+				ReplicaScheduling: dynamicWeightStrategy,
+			},
+			want: []workv1alpha2.TargetCluster{
+				{Name: ClusterMember1, Replicas: 2},
+				{Name: ClusterMember2, Replicas: 2},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
```

Comment:
Can we add the case of having a remainder?

Reply:
done